src/common/m_phase_change.fpp

```diff
@@ -518,6 +524,9 @@ contains
         ! common temperature
         TS = (rhoe + pS - mQ)/mCP
 
+        ! updating maximum number of iterations
+        max_iter_pc_ts = maxval((/ max_iter_pc_ts, ns/))
```

Comment:
Do you also intend for `max_iter_pc_ts` to a variable a user can set higher than `ns` if they so desire? It looks like it because the `maxval` would otherwise be redundant. If so, you should add this as an input variable and distribute it to all ranks.